packages/mux-video/src/index.ts

```diff
@@ -328,6 +330,12 @@ class MuxVideoElement extends CustomVideoElement implements Partial<MuxMediaProp
     }
   }
 
+  get ended() {
+    // This ensures that edge case media that doesn't properly end will
+    // still announce itself as "ended".
+    return getEnded(this.nativeEl, this._hls);
+  }
```

Comment:
👍 I reproduced the `ended` property being false right after the ended event callback so this makes sense.

this would also be easily patched in Media Chrome `media.ended || time > duration - 0.1` or something since we can't shim all the media elements that would be used in Media Chrome.